server/src/weaverbird/backends/pypika_translator/translators/base.py

```diff
@@ -274,6 +275,68 @@ def aggregate(
                 selected_col_names,
             )
 
+    @staticmethod
+    def _pipeline_or_domain_name_or_reference_to_pipeline(
+        pipeline: 'PipelineOrDomainNameOrReference', message: str | None = None
+    ) -> list['PipelineStep']:
+        try:
+            return Pipeline(steps=pipeline).steps
+        except Exception as exc:
+            message = message or f"could not convert {pipeline} to pipeline: {exc}"
+            raise NotImplementedError(message) from exc
+
+    def append(
+        self: Self,
+        *,
+        builder: 'QueryBuilder',
+        prev_step_name: str,
+        columns: list[str],
+        step: "AppendStep",
+    ) -> StepContext:
+        pipelines = [
+            self._pipeline_or_domain_name_or_reference_to_pipeline(pipeline)
+            for pipeline in step.pipelines
+        ]
+        tables: list[str] = []
+        column_lists: list[list[str]] = []
+        for pipeline in pipelines:
+            pipeline_ctx = self.__class__(
+                tables_columns=self._tables_columns, db_schema=self._db_schema_name
+            ).get_query_builder(steps=pipeline, query_builder=builder)
+            tables.append(pipeline_ctx.table_name)
+            column_lists.append(pipeline_ctx.columns)
+            builder = pipeline_ctx.builder
+
+        # By default, UNION ALL will append columns by index in SQL, and only append as much columns
+        # as available in the first SELECT statement. Also, the columns are appended in the order of
+        # the select statement, which is not what we want either (a merge by name is probably what's
+        # expected by the user)
+        columns_to_add = []
+        for column_list in column_lists:
+            for col in column_list:
+                if col not in columns and col not in columns_to_add:
+                    columns_to_add.append(col)
```

Comment:
What about using set here ? instead of looping over columns... 🤔 
```suggestion
         columns_to_add = list((set(columns_to_add).union(set(column_lists))).difference(columns) )
```
What do you think about something like this ? 🤔 


PS : not tested (it may be need to be updated)